File: modules/04_privacy_preserving_ai/code_samples/ios_local_only/LocalOnlyClassifier.swift
// Shows a local-only flow with a clear decision to never call cloud:
struct LocalOnlyClassifier {
    private let coreModel = ImageClassifier() // from Module 3

    func classify(_ image: UIImage) -> (label: String, confidence: Float) {
        // always on-device
        return coreModel.classify(image)
    }
}

File: modules/04_privacy_preserving_ai/code_samples/android_local_only/LocalOnlyClassifier.kt
class LocalOnlyClassifier(private val classifier: Classifier) {

    fun classify(bitmap: Bitmap): ClassificationResult {
        // always on-device
        val res = classifier.classify(bitmap)
        return ClassificationResult(res.label, res.confidence)
    }
}

data class ClassificationResult(val label: String, val confidence: Float)
